### PR TITLE
Resolve some 6 ASAN reports

### PR DESCRIPTION
**Pull request by jengelh**
## Pull Request Type

- [x] Runtime changes
  - [x] Other changes

### Descriptions / 

Continue fixing ASAN reports as they come in.

### Related Issues

https://github.com/DescentDevelopers/Descent3/pull/553

### Checklist

- [x] I have tested my changes locally and verified that they work as intended.
- [x] I have reviewed the changes to ensure they do not introduce any unnecessary complexity or duplicate code.
- [x] I understand that by submitting this pull request, I am agreeing to license my contributions under the project's license.

### CODE REVIEW

File: Descent3/terrain.cpp [hunk 1]
```diff
@@ -293,13 +294,22 @@ void GenerateLODDeltas() {
...
+// The checksum (or rather, an affine modification of it known as
+// occlusion_checksum) is stored in the level's map files, so any modification
+// here ought to not alter the output value, for all imaginable inputs.
 int GetTerrainGeometryChecksum() {
-  int total = 0;
+  unsigned int total = 0;
 
   for (int i = 0; i < TERRAIN_WIDTH * TERRAIN_DEPTH; i++) {
-    total += (Terrain_seg[i].ypos + i);
+    float yi = Terrain_seg[i].ypos + i;
```
Comment (Lgt2x) [inline Descent3/terrain.cpp:305]: I don't get that. Why is `int` + `uint8_t` promoted to float? Is that an obscure C++ gotcha?

File: Descent3/terrain.cpp [hunk 2]
```diff
@@ -293,13 +294,22 @@ void GenerateLODDeltas() {
...
+    if (yi < 2147483648.0f || yi >= 2147483520.0f)
```
Comment (Lgt2x) [inline Descent3/terrain.cpp:306]: Where do these value come from? Is that some IEEE754 magic I'm missing?

File: Descent3/terrain.cpp (hunk 1 above)
Comment (jengelh) [reply Descent3/terrain.cpp]: I must have mixed it up with terrain_segment::y (which is the one that is float). addressed now.

File: Descent3/terrain.cpp (hunk 2 above)
Comment (jengelh) [reply Descent3/terrain.cpp]: The next float number after 2147483520 is 2147483648, but 2147483648 is already too large to be represented in an int32. Does not matter now, code is replaced.

**Comment by jengelh** [comment]
* Rebased (I hope that's ok),
* removed two leftover printfs,
* purged the excessive floating point logic from GetTerrainGeometryChecksum

File: Descent3/aipath.cpp [hunk 3]
```diff
@@ -651,6 +651,8 @@ static inline bool AIPathAddDPathNode(ai_path_info *aip, int *slot, int *cur_nod
     status = AIPathAddDPath(aip, handle);
     if (!status)
       Int3(); // chrishack -- out of dynamic paths
+    if (aip->num_paths < 0)
```
Comment (Lgt2x) [inline Descent3/aipath.cpp:654]: below we have `aip->path_id[aip->num_paths - 1]`, so we should return if `aip->num_paths < 1` instead. Same logic below

**Review by Lgt2x — COMMENTED** [review]
no problem with rebasing branches

**Review by Lgt2x — APPROVED** [review]

**Pull request merged**